components/formats-common/src/loci/common/NIOFileHandle.java

```diff
@@ -263,12 +263,23 @@ public int read(ByteBuffer buf) throws IOException {
   @Override
   public int read(ByteBuffer buf, int off, int len) throws IOException {
```

Comment:
ByteBuffer is limited in size to Integer.MAX_VALUE; see http://docs.oracle.com/javase/6/docs/api/java/nio/ByteBuffer.html